R/app-ui.R

```diff
@@ -71,57 +71,83 @@ app_ui <- function(request) {
+                      )
+                    ),
+                    "Information",
+                    "Select the individual metadata file. This file should have one row per individual, with data about each individual in the experiment. If adding a new dataset to an existing dataset, please include all previous individuals.", # nolint
+                    placement = "top",
+                    trigger = "hover"
+                  )
+                ),
+
+                popify(
                   shinyjs::disabled(
                     fileInput(
-                      "indiv_meta",
-                      "Individual metadata file (.csv)",
+                      "biosp_meta",
+                      "Biospecimen metadata file (.csv)",
                       width = NULL,
                       accept = c(
                         "text/csv",
                         "text/comma-separated-values,text/plain",
                         ".csv"
                       )
                     )
-                  )
+                  ),
+                  "Information",
+                  "Select the biospecimen metadata file. This file should have one row per specimen, with data about each specimen in the experiment. If adding a new dataset to an existing dataset, please include all previous specimens.", # nolint
+                  placement = "top",
+                  trigger = "hover"
                 ),
 
-                shinyjs::disabled(
-                  fileInput(
-                    "biosp_meta",
-                    "Biospecimen metadata file (.csv)",
-                    width = NULL,
-                    accept = c(
-                      "text/csv",
-                      "text/comma-separated-values,text/plain",
-                      ".csv"
-                    )
-                  )
-                ),
 
-                shinyjs::disabled(
-                  fileInput(
-                    "assay_meta",
-                    "Assay metadata file (.csv)",
-                    width = NULL,
-                    accept = c(
-                      "text/csv",
-                      "text/comma-separated-values,text/plain",
-                      ".csv"
+                popify(
+                  shinyjs::disabled(
+                    fileInput(
+                      "assay_meta",
+                      "Assay metadata file (.csv)",
+                      width = NULL,
+                      accept = c(
+                        "text/csv",
+                        "text/comma-separated-values,text/plain",
+                        ".csv"
+                      )
                     )
-                  )
+                  ),
+                  "Information",
+                  "Select the assay metadata file. This file should have one row per specimen, with data about the assay performed on each specimen in the experiment. If adding a new dataset to an existing dataset, please include all previous assay specimens.", # nolint
```

Comment:
"one row per specimen" doesn't seem like it will always be the case (e.g. for behavioral assays)

Reply:
That's a good point. Hmm. Do you have a suggestion on how to clarify this better?

Reply:
Maybe "depending on the assay, this will have one row per specimen or one row per individual (the template will indicate which"